File: library/src/main/java/com/mikepenz/materialdrawer/model/SecondarySwitchDrawerItem.java
package com.mikepenz.materialdrawer.model;

import android.content.Context;
import android.support.annotation.LayoutRes;

import com.mikepenz.materialdrawer.R;
import com.mikepenz.materialdrawer.holder.ColorHolder;

/**
 * Created by mikepenz on 03.02.15.
 */
public class SecondarySwitchDrawerItem extends AbstractSwitchableDrawerItem<SecondarySwitchDrawerItem> {

    @Override
    public int getType() {
        return R.id.material_drawer_item_secondary_switch;
    }

    @Override
    @LayoutRes
    public int getLayoutRes() {
        return R.layout.material_drawer_item_secondary_switch;
    }

    /**
     * helper method to decide for the correct color
     * OVERWRITE to get the correct secondary color
     *
     * @param ctx
     * @return
     */
    @Override
    protected int getColor(Context ctx) {
        int color;
        if (this.isEnabled()) {
            color = ColorHolder.color(getTextColor(), ctx, R.attr.material_drawer_secondary_text, R.color.material_drawer_secondary_text);
        } else {
            color = ColorHolder.color(getDisabledTextColor(), ctx, R.attr.material_drawer_hint_text, R.color.material_drawer_hint_text);
        }
        return color;
    }
}

File: library/src/main/java/com/mikepenz/materialdrawer/model/SecondarySwitchDrawerItem.kt
package com.mikepenz.materialdrawer.model

import android.content.Context
import android.content.res.ColorStateList
import androidx.annotation.LayoutRes
import com.mikepenz.materialdrawer.R
import com.mikepenz.materialdrawer.model.interfaces.IDrawerItem
import com.mikepenz.materialdrawer.util.getSecondaryDrawerIconColor
import com.mikepenz.materialdrawer.util.getSecondaryDrawerTextColor

/**
 * Describes a [IDrawerItem] bing used as secondary item, offering a switch.
 */
open class SecondarySwitchDrawerItem : AbstractSwitchableDrawerItem<SecondarySwitchDrawerItem>() {

    override val type: Int
        get() = R.id.material_drawer_item_secondary_switch

    override val layoutRes: Int
        @LayoutRes
        get() = R.layout.material_drawer_item_secondary_switch

    override fun getColor(ctx: Context): ColorStateList {
        return ctx.getSecondaryDrawerTextColor()
    }

    override fun getIconColor(ctx: Context): ColorStateList {
        return ctx.getSecondaryDrawerIconColor()
    }
}
